### ISSUE TITLE
Support Basic Authorization for custom libvips URLs

### ISSUE THREAD

**Issue by pbomb** [lovell/sharp#938]
There doesn't currently appear to be a mechanism to download the libvips library from a different site using the `SHARP_DIST_BASE_URL` environment variable that also requires BASIC authorization. The `got` request is configured with the following code which doesn't allow for an auth header to be included:
```js
      const gotOpt = {
        agent: caw(null, {
          protocol: 'https'
        })
      };
```
Would you be open to accepting an auth header using an Environment variable like `SHARP_DIST_AUTH_HEADER`? If so, I can submit a pull request.

**Comment by lovell** [comment]
Hello, does it work if you add it to the URL?
```sh
SHARP_DIST_BASE_URL="https://username:password@hostname/path/" npm install sharp
```

**Comment by pbomb** [comment]
Sorry, I should have mentioned that I started with that approach, but the `got` library throws this error: `Error: Basic authentication must be done with auth option`, due to [this issue](https://github.com/sindresorhus/got/pull/114)

**Comment by lovell** [comment]
OK, thanks for checking,

I'm planning to switch from `got` to [simple-get](https://www.npmjs.com/package/simple-get) as sharp will soon add [prebuild-install](https://www.npmjs.com/package/prebuild-install) as a dependency and I'd prefer to use the same library for all HTTP requests.

The `simple-get` package appears to extract basic auth from the URL directly, so that will solve this problem.

**Comment by pbomb** [comment]
Okay, great. Is there a timeline for this? We have a story blocked on this issue currently, so hoping this can get resolved ASAP. Let me know if you'd like me too submit a PR with just that change.

Thanks

**Comment by lovell** [comment]
A PR to swap `got` with `simple-get` would be great, thank you.

**Comment by lovell** [comment]
This will be included in v0.19.0.

**Comment by lovell** [comment]
Included in v0.18.3, thank you for your PR to enable this.
